src/transaction_manager/transaction_status.hpp

```diff
@@ -2,12 +2,45 @@
 
 namespace taraxa {
 
-enum class TransactionStatus {
+enum class TransactionStatusEnum {
   invalid = 0,
   in_block,  // confirmed state, inside of block created by us or someone else
   in_queue_unverified,
   in_queue_verified,
+  executed,
   not_seen
 };
 
-}
+class TransactionStatus {
+ public:
+  TransactionStatusEnum status = TransactionStatusEnum::not_seen;
+  uint32_t period = 0;
+  uint32_t position = 0;
+
+  TransactionStatus() = default;
+
+  TransactionStatus(TransactionStatusEnum transactionStatus, uint32_t tPeriod = 0, uint32_t tPosition = 0) {
+    status = transactionStatus;
+    period = tPeriod;
+    position = tPosition;
+  }
+
+  TransactionStatus(dev::RLP const &rlp) {
+    if (!rlp.isList()) {
+      throw std::invalid_argument("TransactionStatus RLP must be a list");
+    }
+    auto it = rlp.begin();
+    status = (TransactionStatusEnum)(*it++).toInt<uint16_t>();
```

Comment:
I think it will not be `uint16_t`, but `uint8_t`. So safer will be to make `TransactionStatusEnum` a `uint32_t` because we are not saving here anything because of alignment

Reply:
My main concern is how much we save in the database. In the rlp and the db it should be 16bytes or 8 although I would need to check RLP and how it stored data